src/new-components/button/button.test.tsx

```diff
@@ -346,4 +347,22 @@ describe('Button', () => {
             ).toMatchInlineSnapshot(`"⏳"`)
         })
     })
+
+    describe('a11y', () => {
+        it('renders with no a11y violations', async () => {
+            const { container } = render(
+                <>
+                    <Button variant="primary" tone="normal">
+                        Normal
+                    </Button>
+                    <Button variant="primary" tone="destructive">
+                        Destructive
+                    </Button>
+                </>,
+            )
+            const results = await axe(container)
+
+            expect(results).toHaveNoViolations()
+        })
+    })
```

Comment:
There's value in doing more tests for the button in other states: while disabled and while loading? These do affect some a11y related attributes. Maybe you can add those buttons to be rendered in this very test without adding more test cases.

I also think that icon-only buttons deserve their own test case, mirroring this test case, but rendering all buttons as icon-only buttons.